### PR TITLE
[CLEANUP] Make use of PHP 7.1 language features

### PR DESCRIPTION
**Pull request by oliverklee**

### CODE REVIEW

File: config/php-cs-fixer.php [hunk 1]
```diff
@@ -15,6 +15,8 @@
             '@PHP56Migration:risky' => true,
             '@PHP70Migration' => true,
             '@PHP70Migration:risky' => true,
+            '@PHP71Migration' => true,
+            '@PHP71Migration:risky' => true,
```
[the remark this answers is not in the document]
Comment (oliverklee) [reply config/php-cs-fixer.php:19]: The format for the PHP CS Fixer configuration is:
```php
'key of a rule or rule set' => true (always)
```

Some rule sets (including `PHP71Migration`) come in two variants:
- the normal one that will not break anything (unless there's a bug)
- a related set with the `:risky` suffix that includes a certain risk of breaking things

**Comment by oliverklee** [comment]
> Was there a function of PHPStorm to pick all this stuff up? I've no idea if anything was missed, but it seems complete.

I've used a combination of these approaches:
- PHP CS Fixer
- PhpStorm code inspections (including those provided by the PHP Inspections Ultimate plugin)
- read the PHP 7.1 release notes and manually update some things

**Comment by oliverklee** [comment]
> Actually could you keep the `preg_whatever` arguments on separate lines as they were before, where I commented and another place just below. I find it easier to read as it was - particularly having to read two different programming languages (PHP and 'regex').

I have undone these changes and (force-)repushed.

File: config/php-cs-fixer.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (JakeQZ) [reply config/php-cs-fixer.php:19]: > I've used a combination of these approaches:
> 
>     * PHP CS Fixer

Ah, I see.  This picks up a lack of use of PHP 7.1 features and suggests a fix.

File: psalm.xml [hunk 2]
```diff
@@ -2,7 +2,6 @@
 <psalm
     totallyTyped="true"
     strictBinaryOperands="true"
-    requireVoidReturnType="false"
```
Comment (JakeQZ) [inline psalm.xml:5]: Thorough 👍

File: phpcs.xml.dist [hunk 3]
```diff
@@ -84,10 +84,6 @@
     <rule ref="SlevomatCodingStandard.TypeHints.TypeHintDeclaration">
         <!-- Allow PHPDoc with no additional information -->
         <exclude name="SlevomatCodingStandard.TypeHints.TypeHintDeclaration.UselessDocComment"/>
-        <properties>
-            <!-- Void type hint requires PHP 7.1 -->
-            <property name="enableVoidTypeHint" value="false"/>
-        </properties>
```
Comment (JakeQZ) [inline phpcs.xml.dist:90]: Thorough 👍

**Pull request merged**